GAGE5.py

```diff
@@ -0,0 +1,151 @@
+        output_filename += '.fastq'
+
+    with open(output_filename, 'w') as output_file:
+        SeqIO.write(filtered_seqs, output_file, 'fastq')
+
+    return "Filtered data was saved into output file"
+
+
+class BiologicalSequence(ABC):
+    """Abstract base class for biological sequences."""
+
+    def __init__(self, sequence):
+        """Initialize a BiologicalSequence object with a given sequence."""
+        self.sequence = sequence
+
+    def __len__(self):
+        """Return the length of the sequence."""
+        return len(self.sequence)
+
+    def __getitem__(self, index):
+        """Return the item at the specified index."""
+        return self.sequence[index]
+
+    def __str__(self):
+        """Return the string representation of the sequence."""
+        return self.sequence
+
+    @abstractmethod
+    def check_alphabet(self):
+        """Check if the sequence contains valid alphabet characters."""
+        pass
+
+class NucleicAcidSequence(BiologicalSequence):
+    """Abstract base class"""
+
+    DNA_LETTERS = set("ATGCatgc")
+    RNA_LETTERS = set("AUGCaugc")
+    AMINO_ACID_LETTERS = set("ACDEFGHIKLMNPQRSTVWY")
+
+    def check_alphabet(self):
+        """Check if the sequence contains valid nucleic- or aminoacid alphabet characters."""
+        return set(self.sequence).issubset(self.DNA_LETTERS | self.RNA_LETTERS | self.AMINO_ACID_LETTERS)
+
+    def complement(self):
+        """Return the complement sequence."""
+        comp_map_dna = {"A": "T", "G": "C", "T": "A", "C": "G", "a": "t", "t": "a", "g": "c", "c": "g"}
+        return ''.join(comp_map_dna.get(base, base) for base in self.sequence)
+
+    def gc_content(self):
+        """Return the GC content of the sequence."""
+        gc_count = (self.sequence.upper().count('G') + self.sequence.upper().count('C')) / len(self.sequence) * 100
+        return gc_count
+
+class DNASequence(NucleicAcidSequence):
+    """Class representing a DNA sequence."""
+
+    TRANSCRIBE_DICT = {
+        'T': 'U',
+        't': 'u'
+    }
+
+    def __init__(self, sequence):
+        """Initialize a DNASequence object with a given sequence."""
+        super().__init__(sequence)
+        if not self.check_alphabet():
+            raise ValueError("Invalid DNA sequence")
+
+    def transcribe(self):
+        """Transcribe the DNA sequence into an RNA sequence."""
+        transcribed_seq = ''.join(self.TRANSCRIBE_DICT.get(base, base) for base in self.sequence)
+        return transcribed_seq
+
+class RNASequence(NucleicAcidSequence):
+    """Class representing an RNA sequence."""
+
+    def __init__(self, sequence):
+        """Initialize an RNASequence object with a given sequence."""
+        super().__init__(sequence)
+        if not self.check_alphabet():
+            raise ValueError("Invalid RNA sequence")
+
+    def reverse(self):
+        """Return the reverse of the RNA sequence."""
+        return RNASequence(self.sequence[::-1])

```

Comment:
Здесь также, как и в предыдущих случаях, можно возвращать просто развернутую последовательность, а не создавать новый экземпляр класса
```suggestion
        return self.sequence[::-1]
```

Reply:
Ну не совсем. 

У нас был объект сиквенс. Мы его развернули. Почему в ходе разворота тип данных изменила и последовательность превратилась в строку? Хотя тут проблема с тем что захардкожено имя класса, лучше через `type(self)`